spec/paperclip/storage/s3_spec.rb

```diff
@@ -893,10 +1031,12 @@ class DummyCredentialProvider; end
         before do
           object = stub
           @dummy.avatar.stubs(:s3_object).returns(object)
-          object.expects(:write).with(anything,
-                                      content_type: "image/png",
-                                      acl: :public_read,
-                                      cache_control: 'max-age=31557600')
+
+          object.expects((defined?(::Aws) ? :upload_file : :write))
+            .with(anything,
+                  content_type: 'image/png',
+                  acl: Paperclip::Storage::S3::DEFAULT_PERMISSION,
+                  cache_control: 'max-age=31557600')
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.